### PR TITLE
Force exactly one variant in multi-configuration generators

### PR DESCRIPTION
**Pull request by tamiko**
@tjhei Sure.

### CODE REVIEW

**Comment by tamiko** [comment]
@tjhei If you want to test this changes in the testsuite as well, add a corresponding blurbb to
```tests/setup_testsubproject.cmake```

File: cmake/setup_cached_variables.cmake [hunk 1]
```diff
@@ -136,6 +136,17 @@ IF( NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND
 ENDIF()
 
 #
+# Force exactly one build type for multi-cofiguration generators.
+# Unfortunately CMake doesn't support custom configuration types so we just
+# call "DebugRelease" a "Release" build as well.
+#
+IF(CMAKE_BUILD_TYPE MATCHES "Release")
```
Comment (bangerth) [inline cmake/setup_cached_variables.cmake]: I'm going to add the same comment I had previously: this is hard to read. I'd be much happier if this read
```
  IF(CMAKE_BUILD_TYPE MATCHES "(Debug)?Release")
```
or whatever the appropriate syntax is.

File: cmake/setup_cached_variables.cmake [hunk 2]
```diff
@@ -136,6 +136,17 @@ IF( NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND
 ENDIF()
 
 #
+# Force exactly one build type for multi-cofiguration generators.
```
Comment (bangerth) [inline cmake/setup_cached_variables.cmake]: cofiguration -> configuration

**Comment by tamiko** [comment]
@bangerth Yeah, fixed it.

**Pull request closed without merging**